usr.bin/truss/syscalls.c

```diff
@@ -1024,8 +1025,26 @@ get_syscall(struct threadinfo *t, u_int number, u_int nargs)
 		name = new_name;
 	} else
 		new_name = NULL;
+
+	/* Correctly decode compat syscalls arguments by stripping the prefix */
+	if (strncmp("freebsd32_", name, strlen("freebsd32_")) == 0) {
+		lookup_name = name + strlen("freebsd32_");
+	} else if (strncmp("freebsd64_", name, strlen("freebsd64_")) == 0) {
+		lookup_name = name + strlen("freebsd64_");
+	} else {
+		lookup_name = name;
+	}
+
 	STAILQ_FOREACH(sc, &syscalls, entries) {
-		if (strcmp(name, sc->name) == 0) {
+		if (strcmp(lookup_name, sc->name) == 0) {
+			if (name != lookup_name) {
+				/* Print the real syscall name in the logs */
+				struct syscall *new_sc =
+				    malloc(sizeof(struct syscall));
+				*new_sc = *sc;
+				new_sc->name = name;
+				sc = new_sc;
```

Comment:
This makes me a bit nervous with the STAILQ_ENTRY in it.

Reply:
I guess we can skip drop this if we don't want to show the freebsd64/freebsd32 prefix in the output. I think it's vaguely useful to have the prefix for debugging purposes (what kind of executable was running?) but I don't really mind.